### PR TITLE
feat: add options for padding + prefix

### LINKED ISSUE #10: add padding + no prefix options
**Is your feature request related to a problem? Please describe.**
As a user i'd like to have paddingg and no prefix to have better tools for building smart contract calls

### PR DESCRIPTION
**Pull request by shanejonas**
fixes #10

### CODE REVIEW

File: src/serialize/numberToHex.test.ts [hunk 1]
```diff
@@ -4,4 +4,14 @@ describe("numberToHex", () => {
   it("can convert number to hex string", () => {
     expect(numberToHex(16)).toEqual("0x10");
   });
+  it("can pad", () => {
+    const str = "foo";
+    const pad = 10;
+    expect(numberToHex(16, { padStart: pad })).toEqual("0x0000000010");
```
Comment (meowsbits) [inline src/serialize/numberToHex.test.ts]: IMO a control value (`toEqual`) with a clear derivation would be more readable, like `0x + 0.repeat(pad, 10) + 10`.
 ... and using a test value (pad=10) different than an output component (16 hex=> __10__) would be less confusing.

File: tsconfig.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "compilerOptions": {
     "target": "es5",
-    "module": "commonjs",
+    "module": "esnext",
```
Comment (meowsbits) [inline tsconfig.json:4]: How is this related?

Comment (shanejonas) [reply tsconfig.json:4]: its not, but i'd rather this project be on `esnext` so that things like `foo?.bar` work

File: src/serialize/numberToHex.test.ts [hunk 3]
```diff
@@ -4,4 +4,14 @@ describe("numberToHex", () => {
   it("can convert number to hex string", () => {
     expect(numberToHex(16)).toEqual("0x10");
   });
+  it("can pad", () => {
+    const str = "foo";
```
Comment (meowsbits) [inline src/serialize/numberToHex.test.ts]: Unused variable?

File: src/serialize/numberToHex.test.ts [hunk 4]
```diff
@@ -4,4 +4,14 @@ describe("numberToHex", () => {
...
+    const pad = 10;
+    expect(numberToHex(16, { padStart: pad })).toEqual("0x0000000010");
+  });
+  it("can pad with no 0x prefix", () => {
+    const str = "foo";
```
Comment (meowsbits) [inline src/serialize/numberToHex.test.ts]: Unused variable.

File: src/serialize/stringToHex.ts [hunk 5]
```diff
@@ -1,5 +1,18 @@
-const stringToHex = (str: string): string => {
-  return "0x" + new Buffer(str, "utf-8").toString("hex");
+interface IOptions {
+  noPrefix?: boolean;
+  padStart?: number;
+}
+
+const stringToHex = (str: string, options?: IOptions): string => {
+  let prefix = "0x";
+  if (options && options.noPrefix) {
```
Comment (zcstarr) [inline src/serialize/stringToHex.ts:8]: I like a no prefix option :D

**Review by zcstarr — APPROVED** [review]
